src/github.com/getlantern/chained/dialer.go

```diff
@@ -40,9 +40,9 @@ func (d *dialer) Dial(network, addr string) (net.Conn, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Unable to dial server: %s", err)
 	}
-	err = d.sendCONNECT(network, addr, conn)
-	if err != nil {
-		conn.Close()
+	if err := d.sendCONNECT(network, addr, conn); err != nil {
+		// We discard this error, since we are only interested in sendCONNECT
+		_ = conn.Close()
```

Comment:
We need to be explicit that we are ignoring the error for errcheck to pass, which I think is good as well. It reflects what the comment states.